homeassistant/components/surepetcare/__init__.py

```diff
@@ -98,13 +105,25 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     async def handle_set_lock_state(call):
         """Call when setting the lock state."""
-        await spc.set_lock_state(call.data[ATTR_FLAP_ID], call.data[ATTR_LOCK_STATE])
-        await spc.async_update()
+        flap_id = call.data[ATTR_FLAP_ID]
+        state = call.data[ATTR_LOCK_STATE]
+
+        if state == LockState.UNLOCKED.name.lower():
```

Comment:
This would be simpler if you made a dict with LockState.... as key, and the function as data.
Then you could do something like
```
await  LockStateDict[state]
```

Reply:
Good idea, will fix